af/fastpath/fastpath/core.py

```diff
@@ -1248,71 +1276,27 @@ def score_http_requests(msm: dict) -> dict:
     if not reachable:
         scores["blocking_general"] = 1.0
 
-    zzfps = fingerprints["ZZ"]
-    msm_cc = msm.get("probe_cc")
-    ccfps = fingerprints.get(msm_cc, {})
-
     # Scan for fingerprint matches in the HTTP body and the HTTP headers
     # One request is from the probe and one is over Tor. If the latter
     # is blocked the msmt is failed.
     tk = g_or(msm, "test_keys", {})
-    requests = tk.get("requests", []) or []
-    for r in requests:
-        is_tor = gn(r, "request", "tor", "is_tor")
-        body = gn(r, "response", "body")
-        if is_tor is None or body is None:
+    requests = g_or(tk, "requests", [])
+    matches: List[Dict] = []
+    for req in requests:
+        is_tor = gn(req, "request", "tor", "is_tor")
+        resp = g_or(req, "response", {})
+        body = gn(req, "response", "body")
+        match_http_body_fingerprints(resp, matches)
+        match_http_headers_fingerprints(resp, matches)
+
+    if matches:
+        if is_tor:
```

Comment:
I'm not sure this logic is correct.

The `is_tor` key can be set on any individual requests. If it's set you should not flag the whole measurement as inaccurate.

Rather, you should not perform any matching on the fingerprint when it's is_tor, since that's not going to be related to the metadata of the probe itself.

Reply:
Done